warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/handler/edge/evaluation/EdgePreconditionArithmetic.java

```diff
@@ -0,0 +1,295 @@
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+            
+        } else if (right instanceof Set) {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            
+            Iterator iter = ((List) right).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newRight = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.greaterThan(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+        }
+        
+        else {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            if (super.greaterThan(newLeft, newRight)) {
+                addMatchingGroup(newLeft);
+                addMatchingGroup(newRight);
+                matches = true;
+                
+            }
+        }
+        
+        return matches;
+    }
+    
+    @Override
+    public boolean greaterThanOrEqual(final Object left, final Object right) {
+        boolean matches = false;
+        
+        if (left instanceof List && !(right instanceof List)) {
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            Iterator iter = ((List) left).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.greaterThanOrEqual(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+            
+        } else if (right instanceof Set) {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            
+            Iterator iter = ((List) right).iterator();
+            while (iter.hasNext()) {
+                Object tuple = iter.next();
+                Object newRight = Long.parseLong(EventFieldValueTuple.getValue(tuple));
+                if (super.greaterThanOrEqual(newLeft, newRight)) {
+                    addMatchingGroup(tuple);
+                    matches = true;
+                }
+            }
+        }
+        
+        else {
+            Object newLeft = Long.parseLong(EventFieldValueTuple.getValue(left));
+            Object newRight = Long.parseLong(EventFieldValueTuple.getValue(right));
+            
+            if (super.greaterThanOrEqual(newLeft, newRight)) {
+                addMatchingGroup(newLeft);
+                addMatchingGroup(newRight);
+                matches = true;
+                
+            }
+        }
+        
+        return matches;
+    }
+    
```

Comment:
How about the matches(left, right) method?

Reply:
How about that local sports team?  I can add it.